ViAn/Video/video_player.cpp

```diff
@@ -125,23 +143,32 @@ void video_player::convert_frame(bool scale) {
  * @param scale Bool indicating if the frame should be scaled or not.
  * @return Returns the processed frame.
  */
-cv::Mat video_player::process_frame(cv::Mat &frame, bool scale) {
+cv::Mat video_player::process_frame(cv::Mat &src, bool scale) {
     // Copy the frame, so that we don't alter the original frame (which will be reused next draw loop).
-    cv::Mat processed_frame = frame.clone();
+    cv::Mat processed_frame = src.clone();
     processed_frame = video_overlay->draw_overlay(processed_frame, get_current_frame_num());
     if (choosing_zoom_area) {
         processed_frame = zoom_area->draw(processed_frame);
     }
     if (choosing_analysis_area) {
         processed_frame = analysis_area->draw(processed_frame);
     }
-    processed_frame = zoom_frame(processed_frame);
-    processed_frame = contrast_frame(processed_frame);
 
-    if (scale && (frame_width != capture.get(CV_CAP_PROP_FRAME_WIDTH) || frame_height != capture.get(CV_CAP_PROP_FRAME_HEIGHT))) {
-        processed_frame = scale_frame(processed_frame);
-    }
+    zoom_frame(processed_frame).copyTo(processed_frame);
+    contrast_frame(processed_frame).copyTo(processed_frame);

```

Comment:
I changed it primarily for stability. I looked into it further an found that there were no improvements in stability when using `copyTo` so I changed it back.